### ISSUE TITLE
Emojis cut-off in Reactions screen when number of unique reactions is large

### ISSUE THREAD

**Issue by agrawal-d** [zulip/zulip-mobile#3798]
![Screenshot_20200115-015143_Zulip](https://user-images.githubusercontent.com/7714968/72379691-d66f2e80-370b-11ea-811c-89df142d4d7e.jpg)

**Comment by chrisbobbe** [comment]
Aha, you beat me to it! Thanks for opening this issue @agrawal-d as this was exactly what I was testing for.

The problem also occurs on iOS:

![image](https://user-images.githubusercontent.com/22248748/72380677-c3883a00-36ca-11ea-87f1-6a9cfe5c5156.png)

I'll post more soon.

**Comment by chrisbobbe** [comment]
So I see two ways we could fix this:

1) Let the tabs maintain enough width to show the entire emoji, and allow scrolling through the tabs. This would be an easy fix, mainly adding `scrollEnabled: true` in tabBarConfig in the TabNavigatorConfig (https://reactnavigation.org/docs/en/2.x/material-top-tab-navigator.html#tabnavigatorconfig). I've confirmed that this simple config change works on iOS, though I would add a rule to decrease the max width of each tab:

![image](https://user-images.githubusercontent.com/22248748/72381306-0dbdeb00-36cc-11ea-85a2-4a7e275deea9.png)

2) We could remove the tab format and use a SectionList (https://facebook.github.io/react-native/docs/sectionlist) instead, where headers are emoji and items are users.

In either case, I think it would be nice to sort the emoji descending by the number of users, so if 15 people reacted with, e.g., 👍 , and 14 people reacted with 👎 , the 👍 would appear first.

I don't have very strong feelings either way, but one advantage of a SectionList is that it gives a (much) larger area for users to interact with, and it allows interacting from the bottom of the screen, which is easier to reach. This consideration may make up for the downside that the amount of required scrolling increases with the total number of users who have reacted, not just the number of unique reactions.

**Comment by agrawal-d** [comment]
This is how Slack does it:
![image](https://user-images.githubusercontent.com/7714968/72383440-7bf0c500-3741-11ea-91de-d2e69ae3a956.png)

**Comment by agrawal-d** [comment]
During the design process of this feature, the initially proposed design was similar to that of slack (https://github.com/zulip/zulip-mobile/pull/3110)

**Comment by chrisbobbe** [comment]
> During the design process of this feature, the initially proposed design was similar to that of slack (#3110), and would not have faced this issue.

Good point. I also like this idea, from that discussion; it's similar to the SectionList idea I mentioned but would take less vertical space, since the avatars would be listed horizontally:

> An alternative (but here I'm making up something I haven't seen a concrete implementation of) would be a vertical list like this, but each item is a horizontal list of just avatars (like Facebook does in lots of other contexts, I guess.) And when you tap an item to expand, it turns into a vertical list of avatar + name. This idea would need more iterating on, I think.

(from @gnprice )

It looks like the current layout was influenced by Facebook (see this [screenshot](https://github.com/zulip/zulip-mobile/pull/3110#issuecomment-436488128)), but we face a challenge that Facebook doesn't, since Facebook uses a finite number of unique reactions.

**Comment by ray-kraesig** [comment]
> In either case, I think it would be nice to sort the emoji descending by the number of users, so if 15 people reacted with, e.g., [+1], and 14 people reacted with [-1], the [+1] would appear first. This sorting isn't currently done.

Careful with this – it's a good way to make the tabs shuffle around while you're trying to hit them. :sweat:

**Comment by chrisbobbe** [comment]
> Careful with this – it's a good way to make the tabs shuffle around while you're trying to hit them. 😓

Mmm, good point, if we're keeping the tabs. The row of tabs is already a pretty small target to hit, and scroll with, at the top of the screen, which would compound this annoyance with tapping.

If we go the vertical list way (see my previous comment here, and still I think my preference), the shuffling would still affect tap attempts, but it would be noticeably less annoying because there's less of a demand to tap to see who the users are, because you can already see avatars. It's also a much larger area than the row of tabs.

If we're just using a simple SectionList, there is no tapping, and the shuffling would only be a (small, I think?) annoyance with perception, and wouldn't make the user take any unintended actions they would then have to rectify.

I rather like seeing large numbers first, if we can do it in a way that doesn't annoy people. Reactions are ordered this way in the MessageList on a message in the mobile app (though strangely, now I look at it, not in the webapp).

**Comment by sumukhah** [comment]
Hello, can I start working on this?
or is it needs to be discussed further about design?

**Comment by chrisbobbe** [comment]
> Hello, I would like to work on this, is it needs to be discussed further about design?

What are your thoughts on the design discussion so far? 🙂

**Comment by rajprakash00** [comment]
> If we go the vertical list way (see my previous comment here, and still I think my preference), the shuffling would still affect tap attempts, but it would be noticeably less annoying because there's less of a demand to tap to see who the users are, because you can already see avatars. It's also a much larger area than the row of tabs.

@chrisbobbe I'm thinking the design path to be more or less like this 🙂 
Also in this way we don't have to deal with tediuos amount of scrolling for unique reactions(if tab view design is kept) and another good thing is we are also keeping the design in sync with other `SectionList`s used in zulip mobile)(e.g: `UserList` screen)

**Comment by chrisbobbe** [comment]
> I'm thinking the design path to be more or less like this

Sounds good! If you'd like to send a PR with an approach like this, I'll take a look.
